GenericsHomework/GenericsHomework.Tests/NodeTests.cs

```diff
@@ -0,0 +1,65 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace GenericsHomework.Tests
+{
+    [TestClass]
+    public class NodeTests
+    {
+        private (Node<int>,Node<string>) makeNodes()
+        {
+            return (new Node<int>(10), new Node<string>("Miracle Max"));
+        }
+
+        [TestMethod]
+        public void Node_ConstructorIsGeneric()
+        {
+            var (intNode, stringNode) = makeNodes();
+        }
+
+        [TestMethod]
+        public void Node_ToStringWorks()
+        {
+            var (intNode, stringNode) = makeNodes();
+
+            Assert.AreEqual<string>(intNode.ToString(), "10");
+            Assert.AreEqual<string>(stringNode.ToString(), "Miracle Max");
+        }
+
+        [TestMethod]
+       public void Node_Insert_AltersNode()
+        {
+            Node<int> node0 = new(0);
+            Node<int> node1 = new(1);
+
+            node0.Insert(node1);
+         
+            Assert.AreNotEqual<int>(node0.Next.data, node1.Next.data);
+        }
+
+        [TestMethod]
+        public void Node_IsCircular()
+        {
+            Node<int> node0 = new(0);
+            
+            node0.Insert(new(1));
+            node0.Insert(new(2));
+
+            Assert.AreNotEqual<Node<int>>(node0.Next, node0);
+            Assert.AreNotEqual<Node<int>>(node0.Next.Next, node0);
+            Assert.AreEqual<Node<int>>(node0.Next.Next.Next, node0);
+        }
+

```

Comment:
```suggestion
        [TestMethod]
       public void Node_Insert_MultipleNodes()
        {
            // 1 -> 2 -> 3 -> 4
            // ┖──────────────┘

            Node<int> node1 = new(1);
            Node<int> node2 = new(2);
            Node<int> node3 = new(3);
            Node<int> node4 = new(4);

            node1.Insert(node2);
            node2.Insert(node4);
            node2.Insert(node3);
         
            Assert.AreEqual(node1, node3.Next.Next);
            Assert.AreEqual(node2.Next, node3);
            Assert.AreEqual(node1.Next, node2);
            Assert.AreEqual(node3.Next, node4);
        }
```

It would be good to have a few other tests that test for the robustness of your linked list. Making sure that inserting more than two nodes works & the order of insertion doesn't mess up your list. The above code will pass on your current implementation.